types/node/index.d.ts

```diff
@@ -6076,112 +6076,163 @@ declare module "stream" {
             uncork(): void;
         }
 
-        type TransformCallback = (err?: Error, data?: any) => void;
+        type TransformCallback<T> = (err?: Error, data?: T) => void;
 
-        export interface TransformOptions extends DuplexOptions {
-            transform?: (chunk: any, encoding: string, callback: TransformCallback) => any;
-            flush?: (callback: TransformCallback) => any;
+        export interface TransformOptions<TInput = any, TOutput = any> extends DuplexOptions<TInput, TOutput> {
+            transform?: (chunk: TInput, encoding: string, callback: TransformCallback<TOutput>) => void;
+            flush?: (callback: TransformCallback<TOutput>) => any;
         }
 
-        export class Transform extends Duplex {
-            constructor(opts?: TransformOptions);
-            _transform(chunk: any, encoding: string, callback: TransformCallback): void;
+        export class Transform<TInput = any, TOutput = any> extends Duplex<TInput, TOutput> {
+            constructor(opts?: TransformOptions<TInput, TOutput>);
+            _transform(chunk: TInput, encoding: string, callback: TransformCallback<TOutput>): void;
             destroy(error?: Error): void;
         }
 
-        export class PassThrough extends Transform { }
-
-        export function pipeline<T extends NodeJS.WritableStream>(stream1: NodeJS.ReadableStream, stream2: T, callback?: (err: NodeJS.ErrnoException) => void): T;
-        export function pipeline<T extends NodeJS.WritableStream>(stream1: NodeJS.ReadableStream, stream2: NodeJS.ReadWriteStream, stream3: T, callback?: (err: NodeJS.ErrnoException) => void): T;
-        export function pipeline<T extends NodeJS.WritableStream>(stream1: NodeJS.ReadableStream, stream2: NodeJS.ReadWriteStream, stream3: NodeJS.ReadWriteStream, stream4: T, callback?: (err: NodeJS.ErrnoException) => void): T;
-        export function pipeline<T extends NodeJS.WritableStream>(stream1: NodeJS.ReadableStream, stream2: NodeJS.ReadWriteStream, stream3: NodeJS.ReadWriteStream, stream4: NodeJS.ReadWriteStream, stream5: T, callback?: (err: NodeJS.ErrnoException) => void): T;
-        export function pipeline(streams: Array<NodeJS.ReadableStream | NodeJS.WritableStream | NodeJS.ReadWriteStream>, callback?: (err: NodeJS.ErrnoException) => void): NodeJS.WritableStream;
-        export function pipeline(stream1: NodeJS.ReadableStream, stream2: NodeJS.ReadWriteStream | NodeJS.WritableStream, ...streams: Array<NodeJS.ReadWriteStream | NodeJS.WritableStream | ((err: NodeJS.ErrnoException) => void)>): NodeJS.WritableStream;
+        export class PassThrough<T = any> extends Transform<T, T> { }
+
+        export function pipeline<T, TDestination extends NodeJS.WritableStream<T>>(
+            stream1: NodeJS.ReadableStream<T>,
+            stream2: TDestination,
+            callback?: (err: NodeJS.ErrnoException) => void,
+        ): TDestination;
+        export function pipeline<T, A, TDestination extends NodeJS.WritableStream<A>>(
```

Comment:
Perhaps `T1`, `T2`, … rather than switching from `T` to `A`, `B`, … etc. It also aligns the type parameter name with the parameter name (`stream1`, `stream2`, etc.).

Reply:
Done. @rbuckton please let me know if there are any unresolved issues left and maybe you have some thoughts regarding https://github.com/DefinitelyTyped/DefinitelyTyped/pull/26410#discussion_r194250255?